### PR TITLE
CS2100 -> lib_sw_pll development

### PR DESCRIPTION
**Pull request by ed-xmos**
https://xmosjira.atlassian.net/jira/software/c/projects/UA/boards/54?assignee=557058%3A17873fbc-49d6-4971-8efb-a0de95f426a1&selectedIssue=UA-26

Dependent on https://github.com/xmos/lib_xua/pull/360

### CODE REVIEW

File: app_usb_aud_xk_316_mc/src/extensions/audiohw.xc [hunk 1]
```diff
@@ -48,15 +54,30 @@ on tile[0]: in port p_margin = XS1_PORT_1G;  /* CORE_POWER_MARGIN:   Driven 0:
...
 void board_setup()
 {
-    /* "Drive high mode" - drive high for 1, non-driving for 0 */
-    set_port_drive_high(p_ctrl);
 
     /* Ensure high-z for 0.9v */
     p_margin :> void;
 
+// TEMP To enable running on old V1_0 board
+#if V2_0_HARDWARE
```
Comment (xross) [inline app_usb_aud_xk_316_mc/src/extensions/audiohw.xc:62]: We need to make V2 hardware the default behaviour

File: app_usb_aud_xk_316_mc/src/extensions/audiohw.xc [hunk 2]
```diff
@@ -479,6 +504,10 @@ void AudioHwConfig(unsigned samFreq, unsigned mClk, unsigned dsdMode, unsigned s
 
         SetI2CMux(PCA9540B_CTRL_CHAN_0);
     }
+    else if((XUA_SPDIF_RX_EN || XUA_ADAT_RX_EN) && USE_SW_PLL)
+    {
+        /* Do nothing - the SW_PLL configures the AppPLL */
+    }
     else
     {
         AppPllEnable(mClk);
```
Comment (xross) [inline app_usb_aud_xk_316_mc/src/extensions/audiohw.xc:513]: Hopefully this gets pulled into the previous statement with current work to use lib_sw_pll for fixed freqs?

File: app_usb_aud_xk_316_mc/src/extensions/audiohw.xc (hunk 1 above)
Comment (ed-xmos) [reply app_usb_aud_xk_316_mc/src/extensions/audiohw.xc:62]: Good spot - sorry an accidental commit. I'll revert.

File: app_usb_aud_xk_316_mc/Makefile [hunk 3]
```diff
@@ -7,7 +7,7 @@ APP_NAME =
 
 # The flags passed to xcc when building the application
 # The EXTRA_BUILD_FLAGS variable can be used on the xmake command line to add options
-BUILD_FLAGS    = $(EXTRA_BUILD_FLAGS) -fcomment-asm -O3 -report -lquadflash -g -fxscope -DUSB_TILE=tile[0] -DADAT_TX_USE_SHARED_BUFF=1
+BUILD_FLAGS    = $(EXTRA_BUILD_FLAGS) -fcomment-asm -O3 -report -lquadflash -g -fxscope -DUSB_TILE=tile[0] -DADAT_TX_USE_SHARED_BUFF=1 -DUSE_SW_PLL=1
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply app_usb_aud_xk_316_mc/Makefile:10]: Agree - is it OK to address this in the sync mode PR?

[the remark this answers is not in the document]
Comment (ed-xmos) [reply app_usb_aud_xk_316_mc/Makefile:10]: Also will change to XUA_USE_SW_PLL

File: app_usb_aud_xk_316_mc/src/extensions/audiohw.xc (hunk 2 above)
Comment (ed-xmos) [reply app_usb_aud_xk_316_mc/src/extensions/audiohw.xc:498]: Yes

**Review by xross — APPROVED** [review]

File: deps.cmake [hunk 4]
```diff
@@ -1,2 +1,2 @@
-set(APP_DEPENDENT_MODULES "lib_xua"
+set(APP_DEPENDENT_MODULES "ed-xmos/lib_xua(feature/sw_pll)"
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply deps.cmake]: Done

**Pull request merged**